cmd/syncthing/connections.go

```diff
@@ -330,10 +334,10 @@ func (s *connectionSvc) connect() {
 				l.Debugln("Trying relay connections to", deviceID, relays)
 			}
 
-			s.lastRelayCheck[deviceID] = time.Now()
-
 			for _, addr := range relays {
-				uri, err := url.Parse(addr)
+				s.lastRelayCheck[deviceID] = time.Now()
```

Comment:
Right. I guess we could bomb out early:
```
if len(relays) == 0 {
  continue nextDevice
}

s.lastRelayCheck[deviceID] = time.Now()

for relay := range relays {
```

But I am not bothered much, I guess it would just seem less wasteful, and not make me ask questions ;)